### ISSUE TITLE
libjpegturbo 1.5.0 causes streaming to break

### ISSUE THREAD

**Issue by knnniggett** [ZoneMinder/ZoneMinder#1547]
Symptoms:
On a machine with the latest libjpeg-turbo 1.5.0, live streaming is highly intermittent/stutters and the zoneminder log files fill with "Buffer passed to JPEG library is too small"

First reported on Arch Linux here:
https://forums.zoneminder.com/viewtopic.php?p=95165#p95165

I discovered this problem also exists on Fedora 24.

Reproducible: Always
Steps to Reproduce: Install and run ZoneMinder on any machine with libjpeg-turbo 1.5.0.
Temporary Workaround: roll back to libjpeg-turbo 1.4.2

Research:
It seems this is a new "feature" of libjpeg-turbo:
https://github.com/libjpeg-turbo/libjpeg-turbo/issues/78

The commit description provides background on the change they made:
https://github.com/libjpeg-turbo/libjpeg-turbo/commit/68cf83db5647104d9d7ea289c399fe6a5c77da64

TLDR: We are apparently "doing it wrong" when we recycle the cinfo structure in cases when it has been called from other parts of ZoneMinder. I have a fix for this that is incoming.

Additional topic of discussion/possible scope creep:
Upon close inspection of the ZoneMinder source code, I noticed that we have duplicated some of the jpeg library source code, added a "zm_" prefix to the function calls, and placed it into zm_jpeg.cpp. This seems to be the root cause of the problem. Perhaps @mastertheknife could explain why this was done this way, if he is around, since he has done quite a bit of work in this part of the project.
